CSUB-222: e2e test for collect coins

atodorov opened the pull request:
Description of proposed changes:


Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

Commit:
integration-tests: add ethers and ctc vesting contract artifacts

Commit:
integration-tests: deploy ctc script

atodorov committed:
yarn format

atodorov committed:
Deploy CTC contract as part of test setup

0xjeremyfrank commented on integration-tests/src/ethereum/ctc/contracts/GluwaCreditVestingToken.sol/CreditcoinBase.dbg.json:
```diff
@@ -0,0 +1,4 @@
+{
```
Does this work without the "dbg" files? If so, lets remove them.

atodorov replied:
Yes, it works. Removed.

atodorov committed:
Update config values for testnet

- make test unconditional

atodorov committed:
Add test config for devnet

and adjust timeouts

atodorov committed:
Test that secondary call to requestCollectCoins is rejected

and match on the expected error message

atodorov committed:
Make sure to always call .burn() for testing against devnet/testnet

- allows testing to be performed repeatedly
- even if contract has been previously deployed
- doesn't always deploy the contract but simulates actions on mainnet

0xjeremyfrank commented on integration-tests/src/ctc-deploy.ts:
```diff
@@ -3,30 +3,43 @@ import { GluwaCreditVestingToken } from './ethereum/ctc/typechain';
+    let ctcToken : GluwaCreditVestingToken;
+
+    if ( existingAddress !== undefined ) {
+        ctcToken = (await factory.attach(existingAddress)) as GluwaCreditVestingToken;
```
```suggestion
        ctcToken = factory.attach(existingAddress) as GluwaCreditVestingToken;
```

0xjeremyfrank reviewed (approved):
lgtm
